--- ./mcache.c~	2005-10-27 17:58:10.000000000 +0300
+++ ./mcache.c	2005-10-27 17:58:12.000000000 +0300
@@ -222,7 +222,7 @@
   mcache_class_entry_ptr = zend_register_internal_class(&mcache_class_entry TSRMLS_CC);
 
 #ifdef ZTS
-	ts_allocate_id(&mcache_globals_id, sizeof(zend_memcache_globals), (ts_allocate_ctor) php_mcache_init_globals, NULL);
+	ts_allocate_id(&mcache_globals_id, sizeof(zend_mcache_globals), (ts_allocate_ctor) php_mcache_init_globals, NULL);
 #else
 	php_mcache_init_globals(&mcache_globals TSRMLS_CC);
 #endif
